File: Proyecto_1/src/proyecto_olc1/Nodo.java

package proyecto_olc1;

public class Nodo {
    String valor;
    int contHijos;
    Nodo hijos [];
    Nodo hijosT [];
    
    public Nodo(String dato){
        valor = dato;
        this.contHijos=0;
    }
    
    public void redimensionar(){
        hijosT = new Nodo [contHijos+1];
        for (int i = 0; i < this.contHijos; i++) {
            hijosT[i]= hijos[i];
        }
    }
    
    public void aumentarHijo(Nodo nodo){
        redimensionar();
        hijosT[this.contHijos]= nodo;
        hijos = hijosT;
        this.contHijos++;
    }
    
    public String getValor(){
        return valor;
    }
    
    public void setValor(String dato){
        valor = dato;
    }
    
    public void verNodo(){
        System.out.println("{"+valor+" Hijos: "+String.valueOf(contHijos)+"}");
    }
}

File: Proyecto_2/server/src/arbol/nodo.ts

export class Nodo {
    //atributos
    public valor:string;
    public contHijos:number;
    public hijos: Nodo[];
    
    constructor(dato:string){
        this.valor = dato;
        this.contHijos = 0;
        this.hijos = [];
    }
    
    public aumentarHijo(nodo:Nodo):void{
        this.hijos.push(nodo);
        this.contHijos++;
    }
    
    public getValor():string{
        return this.valor;
    }
    
    public setValor(dato:string):void{
        this.valor = dato;
    }
    
    public verNodo():void{
        console.log("{"+this.valor+" Hijos: "+this.contHijos+"}");
    }
}